src/v/storage/opfuzz/opfuzz.cc

```diff
@@ -418,10 +422,41 @@ struct compact_op final : opfuzz::op {
 };
 
 ss::future<> opfuzz::execute() {
-    // execute commands in sequence
-    return ss::do_for_each(_workload, [this](std::unique_ptr<op>& c) {
-        vlog(fuzzlogger.info, "Executing: {}", c->name());
-        return c->invoke(op_context{&_term, &_log, &_as});
+    // compaction operation factory
+    auto compact = [this]() {
+        return ss::do_with(compact_op(), [this](compact_op& c) {
+            return c.invoke(op_context{&_term, &_log, &_as})
+              .handle_exception([](std::exception_ptr e) {
+                  vlog(fuzzlogger.info, "Background compaction error: {}", e);
+              });
+        });
+    };
+
+    return ss::do_for_each(_workload, [this, compact](std::unique_ptr<op>& c) {
+        vlog(
+          fuzzlogger.info,
+          "Executing (with compaction={}): {}",
+          c->concurrent_compaction_safe,
+          c->name());
+
+        std::vector<ss::future<>> ops;
+        ops.push_back(c->invoke(op_context{&_term, &_log, &_as}));
+
+        if (
+          c->concurrent_compaction_safe
+          && random_generators::get_int(0, 100) > 50) {
+            auto f = ss::do_for_each(
+              boost::counting_iterator<size_t>(0),
+              boost::counting_iterator<size_t>(2),
+              [compact](size_t) {
+                  return compact().then([] { return ss::later(); });
```

Comment:
woah! what's `ss::later()!` ... next thing I know we'll get `ss::emoji()`

Reply:
hah. it seems to just guarantee at least one scheduler preemption